README.md

```diff
@@ -148,506 +96,126 @@ export const Label: React.FC<PropsWithChildren<LabelProps>> = ({
-      "class": /** @tw */ "bg-red-500 text-blue-500"
+      class: "font-bold",
     },
-    {
-      "conditions": {
-        fontWeight: "xxl",
-        size: "small"
-      },
-      class: /** @tw */ "bg-gray-500 border-red-500"
-    }
-]
-```
-
-When we do have responsive props what we would do generally is creating an object to all breakpoints you have defined.
-
-Fill the values with their initial values if there is no other responsive props.
-
-So in our use case
-
-```ts
-root({
-  "size": {
-      "initial": "small", // this is mandatory to have an intiial option to respect mobile first approach
-      "md": "large"
-  },
-  "fontWeight": "xxl", // this would not accept a responsive value as you didnt' define it in the responsiveVariants array,
-  className: "awesome-class"
-})
+  ],
+});
 ```
 
-would become
-
-`text-blue-500 text-xs md:text-7xl md:bg-red-500 md:text-blue-500 bg-gray-500 border-red-500 awesome-class`
+## Responsive Variants
 
-5) Complex usage 2. How multiple responsive work with Compound
+To enable responsive variants:
 
-here our definition with compound
+1. Add the variant to the `responsiveVariants` array in your compose function.
+2. Update your Tailwind config to include necessary classes in the safelist.
 
-```ts
-"slots": { 
-  "root": /** @tw */ "text-blue-500",
-  "otherSlot": /** @tw */ "text-blue-500"
-},
-"variants": {
-    "size": {
-        "small": /** @tw */ "text-xs",
-        "large": {
-          root: /** @tw */ "text-7xl",
-          otherSlot: /** @tw */ "text-12xl"
-        }
-    },
-    "fontWeight": {
-        md: /** @tw */ "super-small"
-        xxl: {
-            "root": /** @tw */ "font-extrabold"
-        },
-    }
-},
-"defaultVariants": {
-  "size": "small",
-  "fontWeight": "xxl"
-},
-"responsiveVariants": ["size", "fontWeight"], // note we have added also the fontWeight here
-"compoundVariants": [
-    {
-      "conditions": {
-        "size": "large",
-      },
-      "class": /** @tw */ "bg-red-500 text-blue-500"
-    },
-    {
-      "conditions": {
-        fontWeight: "xxl",
-        size: "small"
-      },
-      class: /** @tw */ "bg-gray-500 border-red-500"
-    }
-]
-```
+```tsx
+// In your variant definition
+export const buttonVariants = compose({
+  // ... other configurations ...
+  responsiveVariants: ["intent"],
+});
 
-Usage:
+// In your Tailwind config
+import { generateSafeList } from "@busbud/tailwind-buddy";
+import { buttonVariants } from "./path-to-your-variants";
 
-```ts
-root({
-  "size": {
-      "initial": "small", // this is mandatory to have an intiial option to respect mobile first approach
-      "md": "large"
-  },
-  "fontWeight": {
-    "initial": "md",
-    "md": "xxl",
-  }, // this would not accept a responsive value as you didnt' define it in the responsiveVariants array,
-  className: "awesome-class"
-})
+export default {
+  // ... other Tailwind configurations ...
+  safelist: generateSafeList([buttonVariants], ["sm", "md", "lg", "xl"]),
+};
```

Comment:
This example would benefit from being more descriptive - what does `"sm", "md", "lg", "xl"` do?